### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Top1Miami**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New method (set a link to paper then)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests or set link to that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ITMO_FS/filters/univariate/measures.py [hunk 1]
```diff
@@ -867,6 +868,185 @@ def modified_t_score(X, y):
...
+    y_der += __mass_star_derivative(X, i, weights, p, radius) * (max(y_lab) + min(y_lab)) / 2
+    return y_der
+
+
+def __update_weights(X, y, alpha, weights, p, radius):
+    return weights + alpha * 2.0 / X.shape[0] * -1 * \
+           np.sum([__y_derivative(X, i, weights, p, radius, y) for i in range(X.shape[0])], axis=0)
+
+
+def weighted_evidential_regression(X, y, alpha=0.01, num_epochs=1000, p=2, k=None, radius=5.0):
```
Comment (LastShekel) [inline ITMO_FS/filters/univariate/measures.py:987]: Does it a filter algorithm? 
Not embedded one?

**Review by LastShekel — CHANGES_REQUESTED** [review]

File: ITMO_FS/filters/univariate/measures.py [hunk 2]
```diff
@@ -867,6 +868,185 @@ def modified_t_score(X, y):
...
+    """
+    Calculates anova measure for each feature.
```
[the remark this answers is not in the document]
Comment (Top1Miami) [reply ITMO_FS/filters/univariate/measures.py:989]: Fixed

File: ITMO_FS/filters/univariate/measures.py (hunk 1 above)
Comment (Top1Miami) [reply ITMO_FS/filters/univariate/measures.py:987]: Yes it is more of an embedded method, but it fits better the pipeline of univariate filter, as it needs cutting rule and provides feature metric that can be used as a rank.

File: ITMO_FS/embedded/WeightedEvReg.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        Examples
+        --------
+        >>> import sklearn.datasets as datasets
+        >>> from ITMO_FS.embedded import WeightedEvReg
+        >>> X = np.array([[1, 2, 3, 3, 1],[2, 2, 3, 3, 2], [1, 3, 3, 1, 3],[3, 1, 3, 1, 4],[4, 4, 3, 1, 5]], dtype = np.integer)
+        >>> y = np.array([1, 2, 3, 4, 5], dtype=np.integer)
+        >>> weighted_ev_reg = WeightedEvReg(cutting_rule=('K best', 2), num_epochs=100)
+        >>> weighted_ev_reg.fit(X, y)
+        >>> print(weighted_ev_reg.selected_features_)
```
Comment (LastShekel) [inline ITMO_FS/embedded/WeightedEvReg.py:55]: do not print this 
and check this test with pytest

**Review by LastShekel — CHANGES_REQUESTED** [review]
Delete pandas from requirements
